trio/__init__.py

```diff
@@ -117,7 +117,7 @@
 
 _deprecate.enable_attribute_deprecations(__name__)
 
-__deprecated_attributes__ = {
+__deprecated_attributes__: "dict[str, _deprecate.DeprecatedAttribute]" = {
```

Comment:
can do `from __future__ import annotations` to not require quoting.

Reply:
Indeed, but since there's only one type hint in the whole module this seemed a little tidier.